awx/job_template_launch_resource.go

```diff
@@ -84,7 +84,11 @@ type JobTemplateLaunchPostRequest struct {
 
 // ExtraVars set a map or external variables sent to the AWX job.
 func (r *JobTemplateLaunchPostRequest) ExtraVars(value map[string]interface{}) *JobTemplateLaunchPostRequest {
-	r.extraVars = value
+	extraVars := make(map[string]int)
+	for key, value := range value {
+		extraVars[key] = value
+	}
+	r.extraVars = extraVars
```

Comment:
what's the desired semantics if you call ExtraVars after having called ExtraVar a few times?
should it append only or replace?

Reply:
Good point. Not clear from the semantics. Two thoughts
- If we revise the current behavior to always append we need to provide a way to clear existing values
- Maybe we can use setters as described [here](https://golang.org/doc/effective_go.html#Getters) and then we can have SetExtraVars that replaces and AddExtraVar that appends

Open for better suggestions...